packages/jumbotron-carousel/edit.js

```diff
@@ -0,0 +1,134 @@
+import Jumbotron from '../jumbotron/edit';
+import classnames from 'classnames';
+
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { Button, Dashicon, PanelBody, Tooltip } from '@wordpress/components';
+import { MediaUpload } from '@wordpress/block-editor';
+
+class JumbotronCarouselEdit extends Jumbotron {
+
+    constructor() {
+        super(...arguments);
+
+        this.onRemoveImage = this.onRemoveImage.bind(this);
+        this.onSelectImage = this.onSelectImage.bind(this);
+    }
+
+    componentDidMount() {
+        const { attributes: { uniqueId }, setAttributes } = this.props;
+        if (!uniqueId) {
+            const uniqueNumber = '' + new Date().getTime() + Math.random();
+            setAttributes({ uniqueId: uniqueNumber });
+        }
+    }
+
+    getHeroClassName() {
+        return classnames(super.getHeroClassName(), 'gb-jumbotron');
+    }
+
+    setHeroBackground() {
+        const { attributes: { carouselImages } } = this.props;
+        var elements = document.getElementsByClassName('gb-hero__background');
+        var image = carouselImages[0] ? carouselImages[0].url : '';
+        [].forEach.call(elements, el =>{
+            el.style.backgroundImage = 'url(\'' + image + '\')';
+        });
+    }
+
+    onRemoveImage(key) {
+        const { attributes: { carouselImages }, setAttributes } = this.props;
+        var images = carouselImages;
+        setAttributes({
+            carouselImages: images.filter(u => u.key !== key)
+        });
+    }
+
+    onSelectImage(img) {
+        const { attributes: { carouselImages }, setAttributes } = this.props;
+        var images = carouselImages;
+
+        setAttributes({ carouselImages: images.concat([{ key: img.id, url: img.url }]) });
+    }
+
+    renderSelectedCarouselImages() {
+        const { attributes: { carouselImages } } = this.props;
+        const elements = [];
+        carouselImages.forEach(image => {
```

Comment:
You could use a `flatMap` instead of a `forEach` and pushing elements to a new array. That way, your elements array would really be immutable.

For this you just have to return an empty array when there is no image url. Otherwise, you return an array with a single jsx component in it.

Reply:
flatMap n'est pas supporter sur IE ou Edge

Reply:
Since we don't need the `if` statement for `.url` check. You can just use the `map` function then.